extract_packets.py

```diff
@@ -25,50 +72,125 @@ def __init__(self, src=None,dst=None,type=None,length=None,
-for packet in packets:
-    
-    packet_analysis = re.search(search_pattern, packet)
-    if(packet_analysis != None):
-        packet_h = Packet(src = packet_analysis.group('src'),
-                          dst = packet_analysis.group('dst'),
-                          type = packet_analysis.group('type'),
-                          length = packet_analysis.group('length'),
-                          nb_segs = packet_analysis.group('nb_segs'),
-                          rss_hash = packet_analysis.group('rss_hash'),
-                          rss_queue = packet_analysis.group('rss_queue'),
-                          hw_ptype = packet_analysis.group('hw_ptype').split(' ') ,
-                          sw_ptype = packet_analysis.group('sw_ptype').split(' ') ,
-                          l2_len = packet_analysis.group('l2_len'),
-                          inner_l2_len = packet_analysis.group('inner_l2_len') ,
-                          l3_len=packet_analysis.group('l3_len'),
-                          inner_l3_len=packet_analysis.group('inner_l3_len'),
-                          l4_len=packet_analysis.group('l4_len'),
-                          receive_queue=packet_analysis.group('receive_queue'),
-                          ol_flags=packet_analysis.group('ol_flags').split(' '))
-                          
-
-
-    # append the packet into the list
-    packets_info.append(packet_h)
-# printing the data in formatted way
-for packet in packets_info:
-    print("The src of the packet :", packet.src, " , The dst : ", packet.dst)
-    print("The type : ", packet.type, ", the length : ", packet.length)
-    print("the nb_segs : ",packet.nb_segs, " ,The rss hash : ", packet.rss_hash, ",The rss queue : ", packet.rss_queue)
-    print("The hw ptype : ", packet.hw_ptype," , The sw ptype : ", packet.sw_ptype)
-    print("l2_len : ", packet.l2_len, " , inner l2_len : ", packet.inner_l2_len)
-    print("l3_len : ", packet.l3_len, " , inner l3_len : ", packet.inner_l3_len)
-    print("l4_len : ", packet.l4_len, " , Receive queue : ", packet.receive_queue)
-    print("ol_flags : ", packet.ol_flags)
-    print("============================================================")
+def open_file(filename) -> str:
+    """open packets file and get the contents of the file
+
+    Parameters
+    -------
+    filename: str
+        is the name of the file that is going to be opened
+        and processed
+
+    Returns
+    -------
+    str
+        the contents of the opened file
+    """
+    with open(filename, encoding="utf-8") as packets_file:
+        return packets_file.read() # Get the contents of the file
+
+
+def split_packets(str_to_process) -> list:
+    """split the passed string into multiple packets
```

Comment:
The spaces in the doc are not right, please check how to run numpy doc validation, and validate your code